tools/src/main/java/org/apache/kafka/trogdor/coordinator/NodeManager.java

```diff
@@ -206,11 +218,19 @@ public void run() {
                 // in the agent's response.  We need to send startWorker requests for these.
                 for (Map.Entry<Long, ManagedWorker> entry : workers.entrySet()) {
                     Long workerId = entry.getKey();
-                    if (!agentStatus.workers().containsKey(workerId)) {
-                        ManagedWorker worker = entry.getValue();
-                        if (worker.shouldRun) {
-                            worker.tryCreate();
-                        }
+                    if (agentStatus.workers().containsKey(workerId))
+                        continue;
+                    ManagedWorker worker = entry.getValue();
+                    if (!worker.shouldRun)
+                        continue;
+
+                    if (!worker.hasExpired()) {
+                        worker.tryCreate();
+                    } else {
+                        log.info("{}: Will not create worker state {} as it has expired. ", node.name(), worker.state);
```

Comment:
This is in the case where a Coordinator re-schedules a task when an Agent is detected via the heartbeats